### PR TITLE
Rename IoT Models to DMR

### PR DESCRIPTION
**Pull request by rido-min**
Use DMR when possible
remove `.ConfigureAwait` from code snippets

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/readme.md [hunk 1]
```diff
@@ -1,9 +1,11 @@
-# IoT Models Repository Samples
+# IoT Models Repository Client Samples
 
-The Azure IoT Models Repository enables builders to manage and share digital twin models for global consumption. The models are [JSON-LD][json_ld_reference] documents defined using the Digital Twins Definition Language ([DTDL][dtdlv2_reference]).
+The Azure IoT Models Repository Client enables builders to manage and share digital twin models for global consumption. The models are [JSON-LD][json_ld_reference] documents defined using the Digital Twins Definition Language ([DTDL][dtdlv2_reference]).
 
 For more info about the Azure IoT Models Repository checkout the [docs][modelsrepository_msdocs].
 
+>Note: The models repository is also referred as DMR (DTDL Models Repository)
```
Comment (drwill-ms) [inline sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/readme.md:7]:
```suggestion
> Note: The models repository is also referred as DMR (DTDL Models Repository)
```

File: sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/readme.md [hunk 2]
```diff
@@ -121,23 +123,23 @@ var client = new ModelsRepositoryClient();
...
-ModelResult result = await client.GetModelAsync(dtmi, ModelDependencyResolution.Disabled).ConfigureAwait(false);
+ModelResult result = await client.GetModelAsync(dtmi, ModelDependencyResolution.Disabled);
 
 // In this case the above dtmi has 2 model dependencies but are not returned
 // due to disabling model dependency resolution.
 Console.WriteLine($"{dtmi} resolved in {result.Content.Count} interfaces.");
 ```
 
-## Digital Twins Model Parser Integration
+## DTDL Parser Integration
```
Comment (drwill-ms) [inline sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/readme.md:133]: Header casing is inconsistent. Above it is Sentence case, but here it is Title Case. Could you standardize it in this document?

**Review by drwill-ms — APPROVED** [review]

**Comment by rido-min** [comment]
@digimaun can you review ?

File: sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/readme.md [hunk 3]
```diff
@@ -1,9 +1,11 @@
-# IoT Models Repository Samples
+# IoT Models Repository Client Samples
 
-The Azure IoT Models Repository enables builders to manage and share digital twin models for global consumption. The models are [JSON-LD][json_ld_reference] documents defined using the Digital Twins Definition Language ([DTDL][dtdlv2_reference]).
+The Azure IoT Models Repository Client enables builders to manage and share digital twin models for global consumption. The models are [JSON-LD][json_ld_reference] documents defined using the Digital Twins Definition Language ([DTDL][dtdlv2_reference]).
```
Comment (digimaun) [inline sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/readme.md:3]: This sentence was meant to give a background on the service that the client is for.

**Comment by digimaun** [comment]
The [sample code](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/ModelsRepositoryClientSamples) that the readme snippets reference need to be in sync as a CI check. For future reference you can update the sample code first, then run [this script](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/modelsrepository/Update-ModelsRepositorySnippets.ps1) to refresh the readme snippets.

File: sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/readme.md (hunk 3 above)
Comment (digimaun) [reply sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/readme.md:3]: Looking at the first sentence in this MS doc about the repository https://docs.microsoft.com/en-us/azure/iot-develop/concepts-model-repository , has very similar verbiage without the usage of "Client".

**Review by digimaun — APPROVED** [review]

**Comment by rido-min** [comment]
seems the PR is approved. who should merge?

**Comment by digimaun** [comment]
> seems the PR is approved. who should merge?

The PR checks need to pass before merging can take place.

**Pull request closed without merging**